src/freenet/support/io/InetAddressIpv6FirstComparator.java

```diff
@@ -21,26 +21,19 @@ public class InetAddressIpv6FirstComparator implements Comparator<InetAddress> {
 	@Override
 	public int compare(InetAddress arg0, InetAddress arg1) {
 		if(arg0 == arg1) return 0;
-		// prefer everything over loopback
+		// prefer everything over loopback / localhost
 		if (!arg0.isLoopbackAddress() && arg1.isLoopbackAddress()) {
 			return -1;
 		} else if (arg0.isLoopbackAddress() && !arg1.isLoopbackAddress()) {
 			return 1;
 		}
-        // prefer LAN routable addresses over link-local addresses
+		// prefer LAN routable addresses over link-local addresses because most of these will not work
 		if (!arg0.isLinkLocalAddress() && arg1.isLinkLocalAddress()) {
 			return -1;
 		} else if (arg0.isLinkLocalAddress() && !arg1.isLinkLocalAddress()) {
 			return 1;
 		}
-		byte[] bytes0 = arg0.getAddress();
-		byte[] bytes1 = arg1.getAddress();
-        // prefer IPv6 over IPv4
-		if(bytes0.length > bytes1.length)
-			return -1;
-		if(bytes1.length > bytes0.length)
-			return 1;
-		// prefer reachable over unreachable. TODO: This actually pings all advertised ip addresses. Is that OK?
+		// prefer reachable over unreachable addresses. This is usually a ping. TODO: This actually pings all advertised ip addresses. Is that OK? Do we need a maximum number of accepted addresses to prevent abuse as DDoS? 
```

Comment:
I don't see this a big avenue for DDoS but we might but a global limit on the physical.udp record itself that is large enough to cover at least X addresses from each address class, but I am not sure what limit this should be as hostnames can be quite large strings and still be valid addresses.

Reply:
I’ll have to look into that, but I don’t think we have to fix that before the next release.